mets_adapter/populate_mets/populate_mets.py

```diff
@@ -60,16 +77,31 @@ def publish(self, space, id):
     prompt="Every record from the storage-service VHS (complete) or just a few (partial)?",
     help="Should this populate from every record in the storage-service VHS?",
 )
-def main(mode):
+@click.option(
+    "--start-record",
+    required=True,
+    type=(str, int),
+    default=(),
+    prompt="Which record do you want to start from (supply id and version)?",
+)
+def main(mode, start_record):
     if mode == "partial":
         num_records = click.prompt("How many records do you want to send?", default=10)
     else:
         num_records = maxsize
-    scanner = StorageManifestScanner()
+
+    scanner = StorageManifestScanner(start_record, num_records)
     publisher = MessagePublisher()
-    for i, (space, id) in zip(range(1, num_records + 1), scanner.scan()):
-        click.echo(f"{click.style(str(i), fg='blue')}: {space}/{id}")
-        publisher.publish(space, id)
+    try:
+        for i, (space, id) in zip(range(1, num_records + 1), scanner.scan()):
```

Comment:
Thought I could have spotted a bug here with it overflowing if `num_records == sys.maxsize`, but apparently not:

```
In [3]: sys.maxsize                                                                                                                                                                                                                                                              
Out[3]: 9223372036854775807

In [4]: sys.maxsize + 1                                                                                                                                                                                                                                                          
Out[4]: 9223372036854775808

In [5]: type(sys.maxsize)                                                                                                                                                                                                                                                        
Out[5]: int

In [6]: type(sys.maxsize + 1)                                                                                                                                                                                                                                                    
Out[6]: int
```

Turns out `sys.maxsize` is not the biggest `int` after all!

Reply:
🙄 of course...

Reply:
https://stackoverflow.com/a/48138759